.github/workflows/sonarqube-scan-pr.yml

```diff
@@ -0,0 +1,25 @@
+on:
+  # Warning: using the pull_request_target event without the cautionary measures may allow
+  # unauthorized GitHub users to open a “pwn request” and exfiltrate secrets.
+  pull_request_target:
+    types: [opened, synchronize, reopened, labeled]
+
+name: SonarCloud Code Scan
+jobs:
+  # For monorepos, create one job per project and specify the directory
+  sonarqube-src:
+    environment: ${{ github.event_name == 'pull_request_target' &&
+      github.event.pull_request.head.repo.full_name != github.repository && 'sonar-pr' || 'none' }}
+    name: scan-synapse-react-client
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          # Disabling shallow clones is recommended for improving the relevancy of reporting
+          fetch-depth: 0
+      - name: SonarQube Scan
+        uses: SonarSource/sonarqube-scan-action@v5.0.0
```

Comment:
Do we want to pin to `v5` to get minor/patch version upgrades automatically, or should we stick with `v5.0.0` for consistency/security?

Reply:
That's a good idea. I took this from SonarCloud docs, but it looks like they now just use `v5`